client/store/review.js

```diff
@@ -3,13 +3,15 @@ import axios from 'axios';
 /* -----------------    ACTION TYPES    ------------------ */
 
 const GET_REVIEWS_FROM_SERVER   = 'GET_REVIEWS_FROM_SERVER';
+const GET_REVIEWS_FOR_PRODUCT   = 'GET_REVIEWS_FOR_PRODUCT';
```

Comment:
This is repetitive to have a new action and action creator.  You can just dispatch GET_REVIEWS_FROM_SERVER from the getReviewsForProductThunkerator with the found reviews.  the new action and action creator do the exact same thing as the old.